src/form.rs

```diff
@@ -44,13 +44,13 @@ pub trait Form {
 	/// The string type.
 	type String: Serialize + PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
 	/// The type identifier type.
-	type TypeId: PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
+	type Type: PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
 	/// A type identifier with indirection.
 	///
 	/// # Note
 	///
 	/// This is an optimization for the compact forms.
-	type IndirectTypeId: PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
+	type IndirectType: PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
```

Comment:
Not sure if we still need `IndirectType` since it was only used for cases where we couldn't use our own `TypeId` as field for a struct to counteract infinitely sized typed, e.g. put them in goo'old `Box` instead. But since our new `TypeId` is always Rust's `core::any::TypeId` we are already safe and don't need this anymore.